### PR TITLE
Get ready for ZK proving

### PR DESCRIPTION
**Pull request by eyusufatik**
# Description
In this PR I modified dependecies and features to make everything work with Risc0 (except evm). 

In the next PR I'll modify features of:
- reth-primitives
- revm-primitives
- revm-precompile

So that we don't have blst lib in our dependency tree, enabling us to run `citrea-stf` in ZkVm

### CODE REVIEW

**Review by kpp — APPROVED** [review]

File: crates/evm/Cargo.toml [hunk 1]
```diff
@@ -30,52 +29,75 @@ jsonrpsee = { workspace = true, features = [
...
+lazy_static = "1.4.0"
+
 
 [features]
-default = ["native"]
+default = []
 native = [
+    "sov-state/native",
+    "sov-modules-api/native",
+
```
Comment (ercecan) [inline crates/evm/Cargo.toml:75]: remove new line

File: crates/evm/Cargo.toml [hunk 2]
```diff
@@ -30,52 +29,75 @@ jsonrpsee = { workspace = true, features = [
...
-    "sov-modules-api/native",
+    "itertools",
+    "serde_json",
+    "secp256k1",
 ]
 serde = []
-smart_contracts = ["native"]
+smart_contracts = [
+    "native",
+
```
Comment (ercecan) [inline crates/evm/Cargo.toml:95]: remove empty line

File: crates/evm/Cargo.toml (hunk 1 above)
Comment (eyusufatik) [reply crates/evm/Cargo.toml:75]: intentional

File: crates/evm/Cargo.toml (hunk 2 above)
Comment (eyusufatik) [reply crates/evm/Cargo.toml:95]: intentional

**Pull request merged**